pkg/controller/worker/firewall_reconcile.go

```diff
@@ -92,6 +92,26 @@ func (a *actuator) ensureFirewallDeployment(ctx context.Context, log logr.Logger
 		},
 	}
 
+	cloudProfileConfig, err := helper.CloudProfileConfigFromCluster(cluster)
+	if err != nil {
+		return err
+	}
+
+	_, partition, err := helper.FindMetalControlPlane(cloudProfileConfig, d.infrastructureConfig.PartitionID)
+	if err != nil {
+		return err
+	}
+
+	controlPlaneConfig, err := helper.ControlPlaneConfigFromClusterShootSpec(cluster)
+	if err != nil {
+		return err
+	}
```

Comment:
Re-reading the cloud profile config can be avoided, it is already done in the `getAdditionalData` func which fetches further resources for both the worker and firewall reconciler.

Reply:
I also stored the partition as it commonly used, too.